app/src/main/java/com/waseefakhtar/doseapp/data/Converters.kt

```diff
@@ -13,11 +16,23 @@ class Converters {
         return value?.let { Date(it) }
     }
 
+    @RequiresApi(Build.VERSION_CODES.O)
+    @TypeConverter
+    fun fromInstant(value: Long?): Instant? {
+        return value?.let { Instant.ofEpochMilli(it) }
+    }
+
```

Comment:
Will this conversion only work for Android O and later? Perhaps add a check and conversion for pre Android O as well?

Reply:
`Instant` has compatibility with API 26 or later. Looking your minSdk, this change may not have sense for now, in my point of view.

Reply:
I agree @Wottrich!